### PR TITLE
Google analytics added

### PR DESCRIPTION
**Pull request by aminata-cisse-fluxtailor**

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -37,3 +37,11 @@
 
 html_static_path = ['_static','OxygenDocs']
 html_css_files = ['css/custom.css']
+googleanalytics_id = "G-8VML9CQJGL"
+
+# Enables the Google Analytics extension when the site is being built in the GitHub Actions environment
+import os
+
+if os.getenv("GITHUB_ACTIONS"):
```
Comment (jonah-bossewitch-fluxtailor) [inline doc/conf.py:45]: Another aside. 

Mature codebases (this is not one yet) often include an easy way to differentiate environments, such as sandbox, dev, stating and prod, often defined in a variable.   We should review this sometime as well https://12factor.net/

**Review by jonah-bossewitch-fluxtailor — CHANGES_REQUESTED** [review]
Hi Aminata - 

Nice work! 🔥 

I left a few inline comments, about venvs w/in the Github action, redundant requirement installs, and different ways to filter out dev traffic.  Probably worth discussing further to see if you have questions, but shouldn't block reworking this PR.

A few other general changes - 

First, we should exclude .DS_Store in .gitignore, like here https://github.com/Flux-Tailor/rate-plan-test-cases/blob/main/.gitignore#L164-L165.   That file a MacOS specific HFS file and should not be commited.  Please add this line to .gitignore and remove these files from the merge.

Next, do you know why the OxygenDocs and input.md files are part of this PR?  Did you add and commit them?  I dont see diffs and am not sure if have a different character for newlines. Did you create your branch directly from main, or from another branch you were working on.  I'm confused why all these files are part of this PR.

Thanks!

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
   docs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+           # We create another virtual environment to isolate our project dependencies
```
[the remark this answers is not in the document]
Comment (aminata-cisse-fluxtailor) [reply .github/workflows/build-docs.yml:14]: Thank you Jonah, that makes a lot of sense!

File: doc/conf.py (hunk 1 above)
Comment (aminata-cisse-fluxtailor) [reply doc/conf.py:45]: Will do, thanks for sharing

**Pull request closed without merging**